ADBDEV-4911-87: Remove redundant parsetree check for NULL.

RekGRpth opened the pull request:
Remove redundant parsetree check for NULL.

At this point, parsetree is always non-NULL, so I removed the redundant
parsetree check for NULL.

red1452 commented on src/backend/executor/spi.c:
```diff
@@ -1947,9 +1947,7 @@ _SPI_prepare_plan(const char *src, SPIPlanPtr plan)
-		if (parsetree == NULL)
-			stmt_list = NIL;
-		else if (plan->parserSetup != NULL)
+		if (plan->parserSetup != NULL)
```
At [line 2150](https://github.com/arenadata/gpdb/pull/732/files#diff-eca689b02874d585dbe0547b5b5af6e723a88247c89b662fa0f601f3d363d4f5R2150) there is the same parsetree pointer. If you remove check here, you should remove check there.

RekGRpth replied:
removed too

RekGRpth committed:
Remove redundant parsetree check for NULL.

At this point, parsetree is always non-NULL, so I removed the redundant
parsetree check for NULL.
